src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/RuntimeImports.cs

```diff
@@ -845,6 +850,307 @@ ushort ClrInstanceID
         [RuntimeImport(RuntimeLibrary, "RhpCheckedXchg")]
         internal static extern object InterlockedExchange([NotNullIfNotNull(nameof(value))] ref object? location1, object? value);
 
+#if TARGET_X86
+        [Intrinsic]
```

Comment:
These methods should not need to be marked as `[Intrinsic]` (both for x86 and !x86). They are not recognized as intrinsics by ryujit nor compiler. It is another left-over from ProjectN that used C/C++ backend and where these methods were treated as C/C++ intrinsics of the same name.

Reply:
Are they not? I think some of such intrinsics in `Math.CoreCLR` are recognized for constant folding and assertion prop, not sure about those here though.

Reply:
They are recognized on Math types, not on RuntimeImports.